addons/binding/org.openhab.binding.exec/README.md

```diff
@@ -39,29 +47,108 @@ The following parameters are automatically added:
-|-----------------|-----------|-----------------------------------------------------------------------------------------|
-| input           | String    | Input parameter to provide to the command                                               |
-| output          | String    | Output of the last execution of the command                                             |
-| exit            | Number    | The exit value of the last execution of the command                                     |
-| run             | Switch    | Send ON to execute the command and the current state tells whether it is running or not |
-| lastexecution   | DateTime  | Time/Date the command was last executed, in yyyy-MM-dd'T'HH:mm:ss.SSSZ format           |
 
+| Channel Type ID | Item Type | Description                                                                          |
+|-----------------|-----------|--------------------------------------------------------------------------------------|
+| input           | String    | Input parameter to provide to the command                                            |
+| output          | String    | Output of the last execution of the command                                          |
+| exit            | Number    | The exit value of the last execution of the command                                  |
+| run             | Switch    | Send ON to execute the command, the current state tells whether it is running or not |
+| lastexecution   | DateTime  | Time/Date the command was last executed, in yyyy-MM-dd'T'HH:mm:ss.SSSZ format        |
 
-**Example**
+
+## Minimal Example
 
 **demo.things**
 
-```
-Thing exec:command:apc [command="/usr/local/bin/apcaccess  status", interval=15, timeout=5]
+```java
+Thing exec:command:apc [command="/usr/local/bin/apcaccess status", interval=15, timeout=5]
 Thing exec:command:myscript [command="php ./configurations/scripts/script.php %2$s", transform="REGEX((.*?))"]
 ```
 
 **demo.items**
 
-```
+```java
 String APCRaw "[%s]" (All) {channel="exec:command:apc:output"}
-Switch APCRunning { channel="exec:command:apc:run"}
-Number APCExitValue {channel="exec:command:apc:exit"}
-DateTime APCLastExecution {channel="exec:command:apc:lastexecution"}
+String APCRunning {channel="exec:command:apc:run"}
+String APCExitValue {channel="exec:command:apc:exit"}
+String APCLastExecution {channel="exec:command:apc:lastexecution"}
 ```
+
+## Full Example
+
+Following is an example how to set up an exec command thing, debug it with a rule and set the returned string to an Number Item. 
+
+**For this to work also the openHAB RegEx Transformation has to be installed**
+
+**demo.things**
+
+```java
+// "%2$s" will be replace by the input channel, this makes it possible to use one command line with different arguments.
+// e.g: "ls" as <YOUR COMMAND> and "-a" or "-l" as additional argument set to the input channel in the rule.
+Thing exec:command:yourcommand [ command="<YOUR COMMAND> %2$s", interval=0, autorun=false ]
```

Comment:
Are you sure it is explicitly needed to be added? I did not do any testing but I'd think the input channel is automatically added to the end of the command during execution!?

Reply:
yes it is. I use it all the time like this. And executing the command without it just executes the command! Without adding the input channel.
https://community.openhab.org/t/openhab2-exec-binding/27819/5?u=josar

Reply:
Okay then, thanks!